### PR TITLE
First Draft

### PR DESCRIPTION
**Pull request by nethad**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,34 @@
...
+NameError: uninitialized constant ObjectStateTest::Character
+    object_state01.rb:19:in `test_01'
+
+1 runs, 0 assertions, 0 failures, 1 errors, 0 skips
+```
+
+So now make this test pass by implementing a `Character` class above the test.
+Only write the code that is **necessary to make the test pass**, nothing more!
+
+After that, continue with number `02`, `03` and so on. After each exercise, you
```
Comment (lxxxvi) [inline README.md]: I think _..."continue with number `02`, `03`"..._ may not clear enough what to do. May be we can list them all, by saying something like:

_Make these commands work:_

* `ruby object_state01.rb` (see example above)
* `ruby object_state02.rb`
* `ruby object_state03.rb`
* `ruby object_state04.rb`
* `ruby object_state05.rb`

File: object_state04.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
+require "minitest/autorun"
+
+################################################################################
+# Extend the class you see here to satisfy the test code.
+#
+# Only add lines, don't change any lines!
+################################################################################
+
+class Character
+  def name=(new_name)
```
Comment (lxxxvi) [inline object_state04.rb]: I think this writer should not exist yet, the student should learn to write it, no?

File: object_state01.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
+require "minitest/autorun"
+
+################################################################################
+# Write yor code below here:
```
Comment (lxxxvi) [inline object_state01.rb]: May be we should explain here in more detail what to do or how to approach it. I think we can move some text from the README here. May be it makes sense that each `object_statexx.rb` has a short explanation what's expected in the current file (only to a certain degree, of course)...

File: object_state01.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+#
+# in your terminal.
+#
+# After you're done, check solutions/solution_01.rb
+# to compare your results.
+#
+################################################################################
+
+class ObjectStateTest < Minitest::Test
+  def self.test_order
```
Comment (lxxxvi) [inline object_state01.rb]: Could we remove this method? I think it's distracting if we ask the students to solve the exercise by reading the tests....

**Review by lxxxvi — COMMENTED** [review]
@nethad 
Very nice work 🐊 . I like the approach of having a simple reader method first, then a writer method, and lastly the constructor 👍 

I added a few thoughts as comments.

I am bit worried about the solutions. I think the explanations in the solutions are super-good and very helpful.
However, since they are solutions, there's a risk that students may give up early and look into solutions. May be we could **emphasize** in the README that they should only look into the solution **after** their implementation worked... and if they're still seeking for a solution, they'd better ask a coach and not look into the solution.

**Comment by lxxxvi** [comment]
Also, we should wait until "Testing" topic has been rewritten, or at least moved to Lesson 10, right?

File: object_state04.rb (hunk 2 above)
Comment (nethad) [reply object_state04.rb]: Good point, might make this a bit more challenging, but I think it makes sense :+1:

File: object_state01.rb (hunk 3 above)
Comment (nethad) [reply object_state01.rb]: I added a few more lines here, but I think it's now clear. Or do you miss something here?

**Comment by nethad** [comment]
Thanks @lxxxvi, I've made a few changes: https://github.com/rubymonstas-zurich/object-state/pull/1/commits/815cdd5b4205b7628538f8596d0b9e2c5037a409

I agree that we should wait for the testing topic, although we're not making it worse right now because other exercises also rely on some testing knowledge, right? We could argue that this makes it even more accessible because the tests are so easy. But of course I agree that we should teach testing first generally.

So I would not wait for it necessarily but at least make sure that we add the testing lesson as soon as possible. What do you think?

File: object_state01.rb (hunk 3 above)
Comment (nerdinand) [inline object_state01.rb]:
```suggestion
# Write your code below here:
```

**Review by nerdinand — APPROVED** [review]
I think this has the potential for some a-ha moments. However it's also one of these exercises that are pretty abstract and require concentration and thinking rather than trying stuff out. Looking forward to seeing how well it works...

File: object_state01.rb [hunk 5]
```diff
@@ -16,16 +16,16 @@
 #
 # in your terminal.
 #
+# Without giving you the solution, the test below should tell you what code we
```
Comment (lxxxvi) [inline object_state01.rb:19]: I like this.

**Review by lxxxvi — APPROVED** [review]
Nice work! Let's see how this works!

**Pull request merged**